libs/api_common/store_helper_internal.cpp

```diff
@@ -15,77 +16,225 @@ ebpf_store_key_t root_registry_key_current_user = HKEY_CURRENT_USER;
 static ebpf_result_t
 _open_ebpf_store_key(_Out_ ebpf_store_key_t* store_key)
 {
+    EBPF_LOG_ENTRY();
+
     // Open root registry path.
     *store_key = nullptr;
 
     // Open the HKCU registry key.
     ebpf_result_t result =
         ebpf_open_registry_key(root_registry_key_current_user, EBPF_STORE_REGISTRY_PATH, KEY_READ, store_key);
 
-    return result;
+    EBPF_RETURN_RESULT(result);
+}
+
+static ebpf_result_t
+_load_extension_header(HKEY data_key, _Out_ ebpf_extension_header_t* extension_header)
+{
+    uint32_t version, size;
+
+    EBPF_LOG_ENTRY();
+
+    ebpf_result_t result = ebpf_read_registry_value_dword(data_key, EBPF_EXTENSION_HEADER_VERSION, &version);
+    if (result != EBPF_SUCCESS) {
+        return result;
+    }
+    result = ebpf_read_registry_value_dword(data_key, EBPF_EXTENSION_HEADER_SIZE, &size);
+    if (result != EBPF_SUCCESS) {
+        return result;
```

Comment:
It looks to me like lines 40 and 44 won't log the exit even though line 36 logged the entry.
Line 50 will but only on success.